spec/session_spec.rb

```diff
@@ -16,9 +16,9 @@
     forbidden_protos = %w( smb tftp imap smtp telnet dict ftp sftp scp file gopher )
     forbidden_protos.each do |forbidden_proto|
       it "should deny a #{forbidden_proto.upcase} request" do
-        @session.base_url = nil
         expect {
-          @session.get('%s://localhost' % forbidden_proto)
+          @session.base_url = '%s://localhost' % forbidden_proto
+          @session.get('')
```

Comment:
Very minor, but why is this change necessary? Do these non-HTTP URLs refuse to work when passed as an argument to `#get` as-is?

Reply:
If `base_url` is already set, it'll be joined with `get` argument.

Reply:
Then I would qualify this as a bug - if the `base_url` specifies, say, `http://something` but we perform `telnet://somewhereelse` explicitly then I'd say what gets supplied in `get` should take priority and override `base_url` for the duration of the call?

Reply:
Yep, I agree on this

Reply:
But let's do it in a separate PR